MapboxNavigation/RouteViewController.swift

```diff
@@ -109,12 +109,65 @@ public class RouteViewController: NavigationPulleyViewController {
                    "or a storyboard reference to Navigation if you are using storyboards.")
     }
     
+    /**
+     Initializes a `RouteViewController` that provides turn by turn navigation
+     for the given route. A optional `direction` object is needed for  potential
+     rerouting.
+
+     See [MapboxDirections.swift](https://github.com/mapbox/MapboxDirections.swift)
+     for further information.
+     */
+    @objc(initWithRoute:directions:)
+    required public init(for route: Route,  directions: Directions = Directions.shared) {
+        let storyboard = UIStoryboard(name: "Navigation", bundle: Bundle.navigationUI)
+        let mapViewController = storyboard.instantiateViewController(withIdentifier: "RouteMapViewController") as! RouteMapViewController
+        let tableViewController = storyboard.instantiateViewController(withIdentifier: "RouteTableViewController") as! RouteTableViewController
+        
+        super.init(contentViewController: mapViewController, drawerViewController: tableViewController)
+        
+        ({
```

Comment:
> The willSet and didSet observers of superclass properties are called when a property is set in a subclass initializer, after the superclass initializer has been called. They are not called  while a class is setting its own properties, before the superclass initializer has been called.

By using the synchronous closure, willSet and didSet observers are getting called properly.
Setting `self.route = route` here is not triggering the didSet observer and therefor not calling `setupRouteController()` w/o the closure. However, we can inline this and call `self.setupRouteController()` manually.